routemaster/webhooks.py

```diff
@@ -0,0 +1,79 @@
+"""Action invocation."""
+
+import enum
+from typing import Any, Dict, List, Callable
+
+import requests
+
+from routemaster.config import Webhook, StateMachine
+
+
+@enum.unique
+class WebhookResult(enum.Enum):
+    """Possible results from invoking a webhook."""
+    SUCCESS = 'success'
+    RETRY = 'retry'
+    FAIL = 'fail'
+
+
+WebhookRunner = Callable[[str, str, bytes], WebhookResult]
+
+
+class RequestsWebhookRunner(object):
+    """
+    Webhook runner which uses `requests` to actually hit webhooks.
+
+    Optionally takes a list of webhook configs to modify how requests are made.
+    """
+
+    def __init__(self, webhook_configs: List[Webhook]=[]) -> None:
```

Comment:
Consider `webhook_configs: Iterable[Webhook]=()` since we only iterate the configs and don't want to accidentally modify the global default.